pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -686,6 +687,48 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 		}
 	}
 
+	// Ensure the install config matches the ClusterDeployment:
+	icSecret := &corev1.Secret{}
+	err = r.Get(context.Background(),
+		types.NamespacedName{
+			Namespace: cd.Namespace,
+			Name:      cd.Spec.Provisioning.InstallConfigSecretRef.Name,
```

Comment:
Do we always expect this to be present? Maybe we should allow for it not to be present before proceeding to validate the install config (which directly calls the unmarshal code)

Reply:
Good point I'll add a check if it's not present to set the condition to indicate there is no install config.